functions.php

```diff
@@ -1385,90 +1385,91 @@ function wp_autoupdates_disable_auto_updates() {
-					update_site_option( 'wp_auto_update_plugins', $wp_autoupdate_plugins );
-
-					/** This filter is documented in wp-admin/includes/class-wp-plugins-list-table.php */
-					$all_plugins           = apply_filters( 'all_plugins', get_plugins() );
-					$wp_autoupdate_plugins = array_intersect( $wp_autoupdate_plugins, array_keys( $all_plugins ) );
-					$enabled_count         = count( $wp_autoupdate_plugins );
-					$plugin_name           = $all_plugins[ $asset ]['Name'];
-
-					$plugin_url = add_query_arg(
-						array(
-							'action'   => 'autoupdate',
-							'plugin'   => $asset,
-							'_wpnonce' => wp_create_nonce( 'autoupdate-plugin_' . $asset ),
-						),
-						'plugins.php'
-					);
-
-					$return_html = sprintf(
-						'<p class="plugin-autoupdate-disabled"><a href="%s" class="edit plugin-autoupdate-enable" aria-label="%s"><span class="plugin-autoupdate-label">%s</span></a></p>',
-						esc_url_raw( $plugin_url ),
-						esc_html__( 'Auto-updates enabled', 'wp-autoupdate' ),
-						esc_html__( 'Enable auto-updates', 'wp-autoupdates' )
-					);
-
-					wp_send_json_success(
-						array(
-							'enabled_count'  => '(' . absint( $enabled_count ) . ')',
-							'disabled_count' => '(' . absint( count( $all_plugins ) - $enabled_count ) . ')',
-							'return_html'    => wp_kses_post( $return_html ),
-							'type'           => 'plugin',
-						)
-					);
-					break;
-				}
-			}
-		}
+		$wp_autoupdate_plugins = (array) get_site_option( 'wp_auto_update_plugins', array() );
+		$wp_autoupdate_plugins = array_diff( $wp_autoupdate_plugins, array( $asset ) );
+		update_site_option( 'wp_auto_update_plugins', $wp_autoupdate_plugins );
+
+		/** This filter is documented in wp-admin/includes/class-wp-plugins-list-table.php */
+		$all_plugins           = apply_filters( 'all_plugins', get_plugins() );
+		$wp_autoupdate_plugins = array_intersect( $wp_autoupdate_plugins, array_keys( $all_plugins ) );
+		$enabled_count         = count( $wp_autoupdate_plugins );
+
+		$plugin_url = add_query_arg(
+			array(
+				'action'   => 'autoupdate',
+				'plugin'   => $asset,
+				'_wpnonce' => wp_create_nonce( 'autoupdate-plugin_' . $asset ),
+			),
+			'plugins.php'
+		);
+
+		$aria_label = esc_attr(
+			sprintf(
+				/* translators: Plugin name. */
+				_x( 'Enable automatic updates for %s', 'plugin', 'wp-autoupdates' ),
+				esc_html( $all_plugins[ $asset ]['Name'] )
+			)
+		);
+
+		$return_html = sprintf(
+			'<p class="plugin-autoupdate-disabled"><a href="%s" class="edit plugin-autoupdate-enable" aria-label="%s"><span class="plugin-autoupdate-label">%s</span></a></p>',
+			esc_url_raw( $plugin_url ),
+			$aria_label,
+			esc_html__( 'Enable auto-updates', 'wp-autoupdates' )
+		);
+
+		wp_send_json_success(
+			array(
+				'enabled_count'  => '(' . absint( $enabled_count ) . ')',
+				'disabled_count' => '(' . absint( count( $all_plugins ) - $enabled_count ) . ')',
+				'return_html'    => wp_kses_post( $return_html ),
```

Comment:
Generally it's not a good idea to inject HTML in the DOM returned by doing AJAX (classic example for XSS). I know WP does this in few places but thinking we shouldn't add another.

Reply:
I agree, but I was trying to make the fewest changes that would fix the issue at hand.   Well, that's not strictly true :-)   I did fix the aria-label and did some refactoring to make those changes easier. 

My current thinking for the merge proposal is to do the AJAX closer to what `WP_Privacy_Requests_Table`  does.  That is, have the PHP that original populates the column output the markup for **all** states and then the JS just hides/shows the relevant markup as it does it's thing.

I think you worked on the original `WP_Privacy_Requests_Table`.  Do you think a similar would be a good idea here?

Reply:
Yeah, seems it would work well here too. Lets look at it in another issue.